src/rust/src/orchard_ffi.rs

```diff
@@ -271,3 +273,47 @@ pub extern "C" fn orchard_batch_validate(batch: *const BatchValidator) -> bool {
         true
     }
 }
+
+#[no_mangle]
+pub extern "C" fn orchard_bundle_outputs_enabled(
+    bundle: *const Bundle<Authorized, Amount>,
+) -> bool {
+    let bundle = unsafe { bundle.as_ref() };
+    bundle.map(|b| b.flags().outputs_enabled()).unwrap_or(false)
+}
+
+#[no_mangle]
+pub extern "C" fn orchard_bundle_spends_enabled(bundle: *const Bundle<Authorized, Amount>) -> bool {
+    let bundle = unsafe { bundle.as_ref() };
+    bundle.map(|b| b.flags().spends_enabled()).unwrap_or(false)
+}
+
+/// Returns whether all actions contained in the Orchard bundle
+/// can be decrypted with the all-zeros OVK.
+///
+/// This should only be called on an Orchard bundle that is
+/// an element of a coinbase transaction.
+#[no_mangle]
+pub extern "C" fn orchard_bundle_has_valid_coinbase_outputs(
+    bundle: *const Bundle<Authorized, Amount>,
+) -> bool {
+    if let Some(bundle) = unsafe { bundle.as_ref() } {
+        for act in bundle.actions() {
+            if try_output_recovery_with_ovk(
+                &OrchardDomain::for_action(act),
+                &OutgoingViewingKey::from([0u8; 32]),
+                act,
+                act.cv_net(),
+                &act.encrypted_note().out_ciphertext,
+            )
+            .is_none()
+            {
+                return false;
+            }
+        }
+    }
+
+    // Either there are no Orchard actions, or all of the outputs
+    // are decryptable with the all-zeros OVK.
+    true
```

Comment:
I agree that true is the right output for the no-bundle case. Currently `HasValidCoinbaseOutputs()` returns false for that case, though.